Adding designs for review for basic VM Edit states

lizsurette opened the pull request:
This includes editing name, description, basic VM details, and allotted Memory and CPU.
https://github.com/lizsurette/ovirt-design/blob/8406b8b1c13d9dd86ba1ce8a4d5aa19723b0d8cd/user-portal/edit-virtual-machine.md

lizsurette committed:
Adding designs for review for basic VM Edit states

This includes editing name, description, basic VM details, and alloted Memory and CPU.

sjd78 commented:
A few questions -

 - Is the assumption that only one card can be put in edit mode at a time?
 - If yes, I assume that:
   - 'Esc' would cancel the Edit
   - Clicking on a different Edit icon would cancel the previous Edit
   - If attempting to close an Edit view that has unsaved changes, a "Are you sure you want to throw changes out" dialog would popup
 - How about a VM with multiple nics and multiple IP addresses (each nic can have >1 IP if you're particularly crazy)?  In webadmin's VM details, each nics has it's own row in a list view.

----

A few comments -
 - **IP Address** and **FQDN** can't be changed from the user portal (or webadmin).  Those are configurations within the VMs OS.

----

Thoughts @lizsurette, @gregsheremeta, @jelkosz, @mareklibra, @bond95?

lizsurette commented:
@sjd78 

> Is the assumption that only one card can be put in edit mode at a time?

This sounds like a good place to start.

> If yes, I assume that:
> 'Esc' would cancel the Edit
> Clicking on a different Edit icon would cancel the previous Edit
> If attempting to close an Edit view that has unsaved changes, a "Are you sure you want to throw changes out" dialog would popup

Yes to all of these.

> How about a VM with multiple nics and multiple IP addresses (each nic can have >1 IP if you're particularly crazy)? In webadmin's VM details, each nics has it's own row in a list view.

I'll think through some options here on scaling.

> IP Address and FQDN can't be changed from the user portal (or webadmin). Those are configurations within the VMs OS.

Makes sense! I will update the mockup.

lizsurette committed:
Updates to basic editing of VM design:
- Moving editing of CPU and RAM from utilization area to the details card.
- Adding concept of next-run configuration changes.

lizsurette commented:
I've made a round of updates.

@sjd78 I'm going to open a new PR around NIC & Disk editing and I will put a proposal for scaling NICs there.

lizsurette committed:
Adding Hot Plug Confirmation modal design.

lizsurette committed:
Removing Add VM documentation. Added by mistake.
Updating documentation for Edit VM.

lizsurette committed:
Updating designs based on feedback, adding link for documentation website.

sjd78 commented:
I'm concerned that the next run configuration changes is going to be a problem.

Take a simple example, the VM name. In webadmin I can edit a VM, change the name (no matter what state the VM is in) and hit 'Ok'.  Another dialog pops up to let me know that my change to 'name' requires a restart.  Clicking 'Ok' there saves the new name (somewhere) and the old name is still displayed on the grid or details view with a new pending changes indicator.

If I edit the VM again, I'll see the new name in the edit box.

On web-ui, we don't currently have a pending changes indicator on the VM list page (at least from what I looked at so far).  But that isn't much of an issue because there is already a flag we can use to throw an indicator on the VM's card.  The issue is that I don't think there is a way that I can currently pull the next run configuration for the VM via the REST api.  I need to verify that (maybe @jelkosz has some insight) further.  Without a way to pull the next run info, all we can do it put a banner on the VM detail page saying "stop and start the VM to apply pending changes so you can see the changes".  (That may not be the best thing to tell a user).

Needs more thought...

lizsurette commented:
@sjd78 Thanks for the write up on the current limitations around next run configuration changes. I think the overall banner is a great first step. Hopefully we can extend this as there is time to make improvements :)

sjd78 commented:
@lizsurette - Verified that I CAN currently access both the current and the next_run information on a VM.  My grep-foo failed me yesterday.

Net result is that if the next_run flag is set on a VM, it will be possible to fetch the next_run data and potentially highlight the things that will change on next shutdown and maybe even show the new value when the edit mode for a card is turned on.

FYI @gregsheremeta

lizsurette committed:
Updating designs based on feedback.

lizsurette committed:
Merge branch 'master' into user-portal-edit-vm
